duration.go

```diff
@@ -122,33 +122,88 @@ func Parse(d string) (*Duration, error) {
 	return duration, nil
 }
 
+// Format formats the given duration into a ISO 8601 duration string (e.g. P1DT6H5M)
+func Format(d time.Duration) string {
+	neg := false
+	if d < 0 {
+		neg = true
+		d = -d
+	}
+
+	if d == 0 {
+		return "PT0S"
+	}
+
+	duration := &Duration{}
+	if d.Hours() >= 8760 {
+		duration.Years = math.Floor(d.Hours() / 8760)
```

Comment:
Would you mind putting these constants into const variables that are package private? I know I didn't do that myself before but it would be nice.

For example 

```Go
const (
    hoursInAYear = 8760
)
```